src/dimensions.jl

```diff
@@ -30,10 +30,20 @@ Return the names of the dimensions for `x`.
     end
 end
 @inline function dimnames(::Type{T}) where {T<:Union{Transpose,Adjoint}}
-    return _transpose_dimnames(dimnames(parent_type(T)))
+    return _transpose_dimnames(Val(dimnames(parent_type(T))))
 end
-_transpose_dimnames(x::Tuple{Symbol,Symbol}) = (last(x), first(x))
-_transpose_dimnames(x::Tuple{Symbol}) = (:_, first(x))
+# inserting the Val here seems to help inferability; I got a test failure without it.
+function _transpose_dimnames(::Val{S}) where {S}
+    if length(S) == 1
+        (:_, first(S))
+    elseif length(S) == 2
+        (last(S), first(S))
+    else
+        throw("Can't transpose $S of dim $(length(S)).")
+    end
+end
```

Comment:
@Tokazama You may want to look at this. I got a test failure on my local master on `@inferred(val_dimnames(y'))` before making this change.